packages/dart/noports_core/lib/src/sshnpd/sshnpd_impl.dart

```diff
@@ -656,21 +731,68 @@ class SshnpdImpl with AtClientBindings, ApkamSigning implements Sshnpd {
 
     // Check if this *client* is allowed connections to the requested host / port
     if (!_permittedToOpen(auth.permitOpen, req)) {
+      await _logEvent(
+        SessionEvent.denied(sessionId: req.sessionId, authInfo: auth.toJson()),
+      );
+
       // Notify noports client that this session is NOT connected
       await _notify(
         atKey: _createResponseAtKey(
           requestingAtsign: requestingAtsign,
           sessionId: req.sessionId,
         ),
-        value: 'Client is not permitted connections to $requested',
+        value:
+            'Connection to $requested denied based on POLICY --permit-open ${auth.permitOpen}',
         sessionId: req.sessionId,
       );
 
       return;
     }
 
+    await _logEvent(
+      SessionEvent.approved(
+        sessionId: req.sessionId,
+        message: 'Connection approved',
+        authInfo: auth.toJson(),
+      ),
+    );
+
     // Start our side of the tunnel
-    await startNpt(requestingAtsign: requestingAtsign, req: req);
```

Comment:
To improve readability and maintainability, voved the try-catch handling out to here instead of within the startNpt function